osf/management/commands/migrate_deleted_date.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import connection, transaction
+from framework.celery_tasks import app as celery_app
+
+logger = logging.getLogger(__name__)
+
+TABLES_TO_POPULATE_WITH_MODIFIED = [
+    'osf_comment',
+    'osf_institution'
+]
+
+POPULATE_COLUMNS = [
+    'SET statement_timeout = 10000; UPDATE osf_basefilenode SET deleted = deleted_on WHERE id IN (SELECT id FROM osf_basefilenode WHERE deleted_on IS NOT NULL AND deleted IS NULL LIMIT %s) RETURNING id;',
+    'SET statement_timeout = 10000; UPDATE osf_abstractnode SET deleted= CASE WHEN deleted_date IS NOT NULL THEN deleted_date ELSE last_logged END WHERE id IN (SELECT id FROM osf_abstractnode WHERE is_deleted AND deleted IS NULL LIMIT %s) RETURNING id;',
+    'SET statement_timeout = 10000; UPDATE osf_privatelink PL set deleted = NL.date from osf_nodelog NL, osf_privatelink_nodes pl_n WHERE NL.node_id=pl_n.abstractnode_id AND pl_n.privatelink_id = pl.id and PL.id in (SELECT id FROM osf_privatelink WHERE is_deleted AND deleted IS NULL LIMIT %s) RETURNING PL.id;',
+]
+
+UPDATE_DELETED_WITH_MODIFIED = 'SET statement_timeout = 10000; UPDATE {} SET deleted=modified WHERE id IN (SELECT id FROM {} WHERE is_deleted AND deleted IS NULL LIMIT {}) RETURNING id;'
+
+
+@celery_app.task(name='management.commands.migrate_deleted_date')
+def populate_deleted(dry_run=False, page_size=1000):
+    with transaction.atomic():
+        for table in TABLES_TO_POPULATE_WITH_MODIFIED:
+            run_statements(UPDATE_DELETED_WITH_MODIFIED, page_size, table)
+        for statement in POPULATE_COLUMNS:
+            run_sql(statement, page_size)
+        if dry_run:
+            raise RuntimeError('Dry Run -- Transaction rolled back')
+
+def run_statements(statement, page_size, table):
+    logger.info('Populating deleted column in table {}'.format(table))
+    with connection.cursor() as cursor:
+        cursor.execute(statement.format(table, table, page_size))
+        rows = cursor.fetchall()
+        if not rows:
+            logger.info('Sentry notification that {} is populated'.format(table))
+
+def run_sql(statement, page_size):
+    table = statement.split(' ')[5]
+    logger.info('Populating deleted column in table {}'.format(table))
+    with connection.cursor() as cursor:
+        cursor.execute(statement, [page_size])
+        rows = cursor.fetchall()
+        if not rows:
+            logger.info('Sentry notification that {} is populated'.format(table))
+
+class Command(BaseCommand):
+    help = '''Populates new deleted field for various models. Ensure you have run migrations
+    before running this script.'''
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--dry_run',
+            type=bool,
+            default=False,
+            help='Run queries but do not write files',
+        )
+        parser.add_argument(
+            '--page_size',
+            type=int,
+            default=1000,
```

Comment:
We might want to bump up the default a bit?  I don't want to lock too much of the table, but I have 250K deleted abstractnodes and almost 2million deleted files (why is this so high)??